frontend/simple/js/idb-flume-log.js

```diff
@@ -0,0 +1,192 @@
+            }
+            current = hash
+            localforage.setItem('current', current, next)
+          })
+        } else {
+          since.set(current)
+          return cb(null, since.value)
+        }
+      }
+      next()
+    })
+  }
+
+  let append = Append(batcher)
+
+  // Pull Stream representig how the log is accessed
+  // API: https://github.com/pull-stream/pull-stream
+  function stream (opts) {
+    opts = opts || {}
+    // Flumedb api for greater than, greater than equal, less than, less than equal
+    let min = opts.gt ? opts.gt : opts.gte ? opts.gte : null
+    let max = opts.lt ? opts.lt : opts.lte ? opts.lte : current
+    let values = opts.values || true
+    let seqs = opts.seqs || false
+    // Only Query inside the parameters of VUEX state
+    let cursor = store ? store.state.logPosition : current
+    let minFound = false
+    let maxFound = false
+
+    return function (end, cb) {
+      if (end) {
+        return cb(end)
+      }
+      // Close if log is empty
+      if (!current) {
+        cb(true)
+      }
+      // close if at the end
+      if (cursor === null) {
+        return cb(true)
+      }
+      let value
+      let seq
+      // Loop through the log until you reach the max (or the most recent)
+      // Throw errors if you the minimum or the end of the list before the maximum\
+      let yeld = () => {
+        if (maxFound && !minFound) {
+          if (value) {
+            if (seqs) {
+              return cb(null, seq)
+            } else {
+              return cb(null, value)
+            }
+          } else {
+            localforage.getItem(cursor, (err, data) => {
+              if (err) {
+                return cb(err)
+              }
+              value = data.data
+              if (cursor === min) {
+                minFound = true
+              }
+              seq = cursor
+              cursor = data.parentHash
+
+              if (seqs) {
+                return cb(null, seq)
+              } else {
+                return cb(null, value)
+              }
+            })
+          }
+        } else {
+          return cb(true)
+        }
+      }
+      let next = (err) => {
+        if (err) {
+          return cb(err)
+        }
+        if (!maxFound) {
+          if (cursor === null) {
+            return cb(true)
+          }
+          localforage.getItem(cursor, (err, data) => {
+            if (err) {
+              return cb(err)
+            }
+            value = data.data
+            if (cursor === min) {
+              minFound = true
+            }
+            if (cursor === max) {
+              maxFound = true
+            }
+            if (!maxFound && minFound) {
+              return cb(new Error('Min is greater than Max in log depth'))
+            }
+            seq = cursor
+            cursor = data.parentHash
+            next()
+          })
+        } else {
+          return yeld()
+        }
+      }
+      next()
+    }
+  }
```

Comment:
I'm guessing the LOC in this function could be reduced by at least half if async/await + Promises were used, which would help me greatly (as the reviewer) and others (including your future self) to understand what's going on. I really actively encourage you (and anyone contributing to the code base) to reduce LOC by embracing async/await. Even if you're not used to it initially, you will love it and wonder how you ever survived without it, or I will buy you lunch/dinner. :P

Same goes for the `get` and `batcher` functions above.